src/Network/Request.php

```diff
@@ -1359,4 +1360,45 @@ public function offsetUnset($name)
     {
         unset($this->params[$name]);
     }
+
+    /**
+     * Returns string representation of the current request in pluginDot syntax
+     *
+     * @params array $options Valid keys are `prefix`, `action`, `pass`, and `query`
+     * @return string Path
+     */
+    public function toPath($options = []) {
+        $prefix = $plugin = $controller = $action = $pass = $query = null;
+        if ($this->plugin) {
+            $plugin = $this->plugin . '.';
+        }
+        if ($prefix = $this->param('prefix')) {
+            $controller = Inflector::camelize($prefix) . '/';
+        }
+        $controller .= $this->param('controller') . 'Controller';
+
+        if (!empty($options['action'])) {
+            $action = '/' . $this->param('action');
+        }
+
+        if (!empty($options['pass']) && $passedArgs = $this->param('pass')) {
+            $pass = '/' . implode('/', $passedArgs);
+        }
+
+        if (!empty($options['query']) && $this->query) {
+            $query = '?' . http_build_query($this->query);
+        }
+
+        return $plugin . $controller . $action . $pass . $query;
```

Comment:
I'm bad at naming things, so the `toPath()` or `toActionPath()` is just a proposal.

Regarding fullynamespace classname + method, I feel it's a bit too long.  I'm after a syntax that is short and easy to use in config or bootstrap files.

Hmm, I guess this feature is too app specific judging from the lack of interest. I will close this as this can easily be implemented out of core.